src/apps/orders/views.py

```diff
@@ -0,0 +1,105 @@
+from django.contrib.auth.mixins import UserPassesTestMixin, LoginRequiredMixin
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views import View
+from django.views.generic import DetailView
+
+from src.apps.basket.basket import Basket
+from .forms import DeliveryOptionForm, PaymentMethodForm
+from .models import Order, OrderProduct
+
+
+class CheckoutView(LoginRequiredMixin, View):
+    template_name = "orders/checkout.html"
+
+    def get(self, request, *args, **kwargs):
+        basket = Basket(request)
+        total_price = basket.get_total_price()
+        delivery_option_form = DeliveryOptionForm()
+        payment_method_form = PaymentMethodForm()
+
+        return render(
+            request,
+            self.template_name,
+            {
+                "basket": basket,
+                "total_price": total_price,
+                "delivery_option_form": delivery_option_form,
+                "payment_method_form": payment_method_form,
+            },
+        )
+
+    def post(self, request, *args, **kwargs):
+        basket = Basket(request)
+        total_price = basket.get_total_price()
+        delivery_option_form = DeliveryOptionForm(request.POST)
+        payment_method_form = PaymentMethodForm(request.POST)
+
+        if delivery_option_form.is_valid() and payment_method_form.is_valid():
+            order = Order.objects.create(
+                user_id=request.user,
+                user_name=request.user.first_name,
+                user_email=request.user.email,
+            )
+
+            for item in basket:
+                product = item["product"]
+                quantity = item["quantity"]
+                price = item["price"]
+                OrderProduct.objects.create(
```

Comment:
Using bulk_create is recommended when you need to insert multiple records at once, as it minimizes the number of database queries, which can significantly improve performance